docs/versioning.md

```diff
@@ -0,0 +1,101 @@
+Per the rules of SemVer 2.0, during the initial development of ONNX, we use a MAJOR version of 0 for both the IR version and operator version and will only increment the MINOR version in the face of either a breaking change or the need to stabiize a version for specific engineering purposes. [ISSUE: decide how we will stabilize and archive released versions of ONNX] 
+
+### Serializing SemVer version numbers in protobuf
+
+For historical reasons, ONNX serializes the MAJOR, MINOR, and PATCH values as a bit-packed 32-bit integer; the most siginificant byte is the MAJOR component, the second most significant byte is the MINOR component, the least significant two bytes are the PATCH component. 
+
+For example, 1.2.3 is represented as 0x01020003.
+
+The pre-release and build metadata are represented as ISSUE: are we supporting these or not, and if so, on what entity types? 
+
+
+
+
+
+## IR versioning
+
+Changes to the file format or graph model semantics version atomically. Breaking changes to the format or semantics of the ONNX specification require an increment of the MAJOR version.  Non-breaking format or semantic changes that introduce new functionality require an increment of the MINOR version. Non-breaking changes to the specification that simply clarify spec ambiguities require an increment of the PATCH version.  
+
+The ONNX IR format adheres to the versioning guidelines defined in the [Updating a Message Type](https://developers.google.com/protocol-buffers/docs/proto3#updating) section of the proto3 specification.  
+
+As a general principle, implementations SHOULD be robust in the face of missing fields. However, to ensure basic interoperation, a subset of message fields will be marked as required for a given IR version and all producers MUST set these fields correctly. Required fields always are marked with the comment:
+
+    // This field MUST be present for this version of the IR.
+
+By way of example, the ModelProto.ir_version MUST be present in every model.
+
+ISSUE: decide and document how we want to handle changes to the format that are forward compatible but will cause the generated code to introduce a breaking change (e.g., changing the data type of a field from int64 to int32, int64 to double or bytes to string).  I would recommend that we never do it after we hit 1.0.0 - prior to that we should be fairly liberal with changes to size (e.g., int32<>int16, float<>double) and conservative with changes to value space (e.g., integral<>floatingpoint, string<>int, scalar<>message)
+
+ISSUE: define type compatiblity rules either here or under model versioning - probably here
+
+## Operator versioning
+
+ISSUE: sort out the design and then document.
+
+## Model versioning
+
+Model authors and applications/systems  MAY elect to ignore model versioning mechanism and policy rules. For models that will be shared across developers, teams, or organizations, model authors and applications/systems SHOUD adhere to the following version policies:
```

Comment:
if we don't provide any tooling for handling model versioning we might as well just say that it's up to using application to handle. In my opinion it really depends on the particular use case of onnx and it'd be different in different organizations

Reply:
That was my very first sentence, which makes this section non-normative.

I think having us collectively document how we THINK it should work is strictly goodness.

I also think that some of this is text that we will need for operators, since an operator is just a graph who's nodes are opaque :-)

Reply:
That stated, I added more prose to make that clear.